packages/sdk/src/lib/IExecDataProtectorModule.ts

```diff
@@ -0,0 +1,46 @@
+import { Eip1193Provider } from 'ethers';
+import { GraphQLClient } from 'graphql-request';
+import { IExec } from 'iexec';
+import {
+  DEFAULT_CONTRACT_ADDRESS,
+  DEFAULT_IEXEC_IPFS_NODE,
+  DEFAULT_IPFS_GATEWAY,
+  DEFAULT_SUBGRAPH_URL,
+} from '../config/config.js';
+import { AddressOrENS, Web3SignerProvider } from './types/commonTypes.js';
+import { DataProtectorConfigOptions } from './types/dataProtectorTypes.js';
+
+class IExecDataProtectorModule {
```

Comment:
We can probably set this class as `abstract` to disallow instantiation, ie. you can only instantiate IExecDataProtector.

```suggestion
abstract class IExecDataProtectorModule {
```

Reply:
you are right